src/services/ServiceLocator.ts

```diff
@@ -0,0 +1,57 @@
+/*-
+ *
+ * Hedera Local Node
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { IService } from './IService';
+
+export class ServiceLocator {
+    private services: Map<string,IService> = new Map<string, IService>();
+
+    public static Current: ServiceLocator;
+
+    public static Initiailze(): void {
```

Comment:
Not sure but I think if this need to be singleton the method should be private?

Reply:
We use it in the Bootstrapper to initiliaze the Service Locator the first time.